File: app/src/main/java/com/pr0gramm/app/ui/MainActionHandler.java
package com.pr0gramm.app.ui;

import android.os.Bundle;
import android.support.annotation.Nullable;

import com.pr0gramm.app.feed.FeedFilter;
import com.pr0gramm.app.ui.fragments.ItemWithComment;

/**
 */
public interface MainActionHandler {
    void onLogoutClicked();

    void onFeedFilterSelected(FeedFilter filter);

    void onFeedFilterSelected(FeedFilter filter, @Nullable Bundle searchQueryState);

    void onFeedFilterSelected(FeedFilter filter, @Nullable Bundle searchQueryState,
                              @Nullable ItemWithComment startAt);

    void pinFeedFilter(FeedFilter filter, String title);

    void showUploadBottomSheet();

}

File: app/src/main/java/com/pr0gramm/app/ui/MainActionHandler.kt
package com.pr0gramm.app.ui

import android.os.Bundle
import com.pr0gramm.app.feed.FeedFilter
import com.pr0gramm.app.ui.fragments.CommentRef

/**
 */
interface MainActionHandler {
    fun onLogoutClicked()

    fun onFeedFilterSelected(filter: FeedFilter)

    fun onFeedFilterSelected(filter: FeedFilter, searchQueryState: Bundle?)

    fun onFeedFilterSelected(filter: FeedFilter, queryState: Bundle?,
                             startAt: CommentRef?, popBackstack: Boolean = false)

    fun pinFeedFilter(filter: FeedFilter, title: String)

    fun showUploadBottomSheet()

}
